packages/patternfly-4/react-core/src/components/Tabs/Tabs.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Tabs/tabs.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import { AngleLeftIcon, AngleRightIcon } from '@patternfly/react-icons';
+import { getUniqueId, isElementInView, sideElementIsOutOfView } from '../../internal/util';
+import { SIDE } from '../../internal/constants';
+
+const propTypes = {
+  /** content rendered inside the Tabs Component. */
+  children: PropTypes.node.isRequired,
+  /** additional classes added to the Tabs */
+  className: PropTypes.string,
+  /** the index of the active tab */
+  activeKey: PropTypes.number,
+  /** handel tab selection */
+  onSelect: PropTypes.func,
+  /** Enables the filled tab list layout */
+  isFilled: PropTypes.bool,
+  /** Enables Secondary Tab styling */
+  isSecondary: PropTypes.bool,
+  /** Aria Label for the left Scroll Button */
+  leftScrollAriaLabel: PropTypes.string,
+  /** Aria Label for the right Scroll Button */
+  rightScrollAriaLabel: PropTypes.string
+};
+
+const defaultProps = {
+  className: '',
+  activeKey: 0,
+  onSelect: () => undefined,
+  isFilled: false,
+  isSecondary: false,
+  leftScrollAriaLabel: 'Scroll left',
+  rightScrollAriaLabel: 'Scroll Right'
+};
+
+class Tabs extends React.Component {
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+
+  constructor(props) {
```

Comment:
Since this is a class component & we are using the babel class properties plugin, we don't need a constructor. You can add the createref right on the class like tablist= React.createRef()